libraries/GCS_MAVLink/GCS_Common.cpp

```diff
@@ -350,7 +350,7 @@ void GCS_MAVLINK::send_battery_status(const uint8_t instance) const
                                     current,      // current in centiampere
                                     consumed_mah, // total consumed current in milliampere.hour
                                     consumed_wh,  // consumed energy in hJ (hecto-Joules)
-                                    percentage,
+                                    constrain_int16(percentage, 0, 100),
```

Comment:
These functions are a bit tricky; you do need to use the right one for the input type.  In this case you should be using `constrain_int8`.

Of course, that function doesn't actually exist.  So I guess this is near enough...

Reply:
oh yeah, -1 is possible there. I'll fix that up.

There is no constrain_int8().. are you suggesting I create it?